### DISCUSSION TITLE
Feature request: Lifecycle hooks in LangFuse

### DISCUSSION THREAD

**Discussion by barseghyanartur** [langfuse/langfuse#4342] [Ideas]
### Describe the feature or potential improvement

Introduce lifecycle hooks in LangFuse to trigger custom actions during key events (such as prompt promotion to production).

These hooks should provide flexibility, allowing integration with external systems via webhooks.

Initial implementation could be as simple as managing that through the UI:

- Event to hook into.
- Webhook URL.
- Configuration (security tokens, retries, timeouts).
- Additional payload parameters.

On a long term, support for a reproducible configuration format for defining lifecycle hooks would be useful. Think of version-controlled deployments or “manual” environment duplications.

**Comment by marcklingen** [comment]
Thanks for sharing! I completely agree that this would be helpful in many ways for expanding around Langfuse.

Do you think this thread reflects your idea well enough? https://github.com/orgs/langfuse/discussions/1033

I'd love to merge the conversations if possible.

**Reply by barseghyanartur** [reply]
I think, in the essence - yes. It might make sense to expand the list of potential events in that discussion.

**Reply by barseghyanartur** [reply]
I left a comment in the discussion you suggested. Closing this one.

**Reply by marcklingen** [reply]
thank you!
